app/src/org/commcare/location/CommCareProviderLocationController.kt

```diff
@@ -0,0 +1,63 @@
+package org.commcare.location
+
+import android.content.Context
+import android.location.Location
+import android.location.LocationListener
+import android.location.LocationManager
+import android.os.Bundle
+import org.commcare.utils.GeoUtils
+
+/**
+ * @author $|-|!˅@M
+ */
+class CommCareProviderLocationController(private val mContext: Context,
+                                         private val mListener: CommCareLocationListener): CommCareLocationController {
+
+    private val mLocationManager = mContext.getSystemService(Context.LOCATION_SERVICE) as LocationManager
+    private var mCurrentLocation: Location? = null
+    private val mProviders = GeoUtils.evaluateProviders(mLocationManager)
+    private val mLocationListener = object: LocationListener {
+        override fun onLocationChanged(location: Location?) {
+            location ?: return
+            mCurrentLocation = location
+            mListener.onLocationResult(mCurrentLocation!!)
+        }
+
+        override fun onStatusChanged(provider: String?, status: Int, extras: Bundle?) {
+            //This callback will never be invoked.
+        }
+
+        override fun onProviderEnabled(provider: String?) {
+            TODO("Not yet implemented")
+        }
+
+        override fun onProviderDisabled(provider: String?) {
+            TODO("Not yet implemented")
```

Comment:
should we not communicate this to `mListener` so that it can take actions in case the provider disabled ?

Reply:
Maybe yeah! I just copied over our previous code for this and it wasn't implemented.

Reply:
yeah we should check for this since the purpose of this work is to make this feature more robust.

Reply:
I've been thinking about this, and it feels like these 2 methods are not needed!

Reply:
@ShivamPokhriyal It would be great to append your statements with some reasoning! It's really hard to understand what you have been thinking otherwise.

Reply:
Yeah sorry about that. So, the onProviderDisabled update will come when one of the 2 providers we're using(GPS and network) gets disabled, which simply means that the location updates will cease for that provider, and as soon as it gets enabled again, the updates will resume.  

And we already have a check [in place ](https://github.com/dimagi/commcare-android/pull/2282/files/820f167f3e358de42212fc5df23c56f9792a24bb#diff-784d69745186464f14c4e514ad19ef93R45) which ensures that if the providers are already disabled, we send the user to the appropriate settings page.

Reply:
@ShivamPokhriyal  Thanks for the explanation. What if the user stops the provider after our initial check ? would not we want to show the error to the user in that case ?

Reply:
I updated the code to use the broadcast receiver from `PollSensorController` here only. So now, we'll listen to providers change events in the broadcast receivers and these methods can be ignored fully.